examples/tutorials/nb_python/ECCV_2020_Navigation.py

```diff
@@ -0,0 +1,1130 @@
+#   - **edge_max_len** - The maximum allowed length for contour edges along the border of the mesh. Extra vertices will be inserted as needed to keep contour edges below this length. A value of zero effectively disables this feature. [Limit: >=0] [ / cell_size]
+#   - **edge_max_error** - The maximum distance a simplfied contour's border edges should deviate the original raw contour. [Limit: >=0]
+#   - **verts_per_poly** - The maximum number of vertices allowed for polygons generated during the contour to polygon conversion process.[Limit: >= 3]
+#   - **detail_sample_dist** - Sets the sampling distance to use when generating the detail mesh. (For height detail only.) [Limits: 0 or >= 0.9] [x cell_size]
+#   - **detail_sample_max_error** - The maximum distance the detail mesh surface should deviate from heightfield data. (For height detail only.) [Limit: >=0] [x cell_height]
+#
+#
+#
+#
+
+# %%
+# @markdown ## Recompute NavMesh:
+navmesh_settings = habitat_sim.NavMeshSettings()
+
+# @markdown Choose Habitat-sim defaults (e.g. for point-nav tasks), or custom settings.
+use_custom_settings = True  # @param {type:"boolean"}
+sim.navmesh_visualization = True  # @param {type:"boolean"}
+if not use_custom_settings:
+    navmesh_settings.set_defaults()
+else:
+    # @markdown ## Configure custom settings (if use_custom_settings):
+    # @markdown Configure the following NavMeshSettings for customized NavMesh recomputation.
+    # @markdown **Voxelization parameters**:
+    navmesh_settings.cell_size = (
+        0.05
+    )  # @param {type:"slider", min:0.01, max:0.2, step:0.01}
+    # default = 0.05
+    navmesh_settings.cell_height = (
+        0.2
+    )  # @param {type:"slider", min:0.01, max:0.4, step:0.01}
+    # default = 0.2
+
+    # @markdown **Agent parameters**:
+    navmesh_settings.agent_height = (
+        1.5
+    )  # @param {type:"slider", min:0.01, max:3.0, step:0.01}
+    # default = 1.5
+    navmesh_settings.agent_radius = (
+        0.1
+    )  # @param {type:"slider", min:0.01, max:0.5, step:0.01}
+    # default = 0.1
+    navmesh_settings.agent_max_climb = (
+        0.2
+    )  # @param {type:"slider", min:0.01, max:0.5, step:0.01}
+    # default = 0.2
+    navmesh_settings.agent_max_slope = (
+        45
+    )  # @param {type:"slider", min:0, max:85, step:1.0}
+    # default = 45.0
+
+    # @markdown **Navigable area filtering options**:
+    navmesh_settings.filter_low_hanging_obstacles = True  # @param {type:"boolean"}
+    # default = True
+    navmesh_settings.filter_ledge_spans = True  # @param {type:"boolean"}
+    # default = True
+    navmesh_settings.filter_walkable_low_height_spans = True  # @param {type:"boolean"}
+    # default = True
+
+    # @markdown **Detail mesh generation parameters**:
+    # @markdown For more details on the effects
+    navmesh_settings.region_min_size = (
+        20
+    )  # @param {type:"slider", min:0, max:50, step:1}
```

Comment:
Black broke all these sliders... hmm... we may need to run off the fmt manually with ```# fmt : off```

see black's documentation. NB: @aclegg3

Reply:
If this doesn't work, we may have to build a quick shim with ipython widgets.

Reply:
This part belongs to @aclegg3 . He may need to take a look.

Reply:
I thought we were trying to avoid the resultant python files in black anyway, because of the colab decorators that get moved around.

Reply:
Yeah, those sections should be marked off with `#fmt: off/on` for that reason @jturner65